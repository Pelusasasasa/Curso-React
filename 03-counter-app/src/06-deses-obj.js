
// Desestructuración
// Asignación Desestructurante
const persona = {
    nombre: 'Tony',
    edad: 45,
    clave: 'Ironman'
};

export const usContext = ({ clave, nombre, edad, rango = 'Capitán' }) => {
    return {
        nombreClave: clave,
        edad,edad,
        latlng: {
            lat: 14.1232,
            lng: -12.3232
        },
        nombre,
        rango
    }

}

const { nombreClave, anios, latlng: { lat, lng } } = usContext( persona );


