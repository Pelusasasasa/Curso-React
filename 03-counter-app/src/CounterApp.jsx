import PropTypes from 'prop-types';
import { useState } from 'react';
 
export const CounterApp = ({value}) => {
    
    const [count,setCount] = useState(value);

    const handleAdd = (e) => {
        setCount(count + 1);
    }

    const handleSubstract = (e) => {
        setCount(count - 1);
    }

    const handleReset = (e) => {
        setCount(value);
    };

  return (
    <>
        <h1> Counter App </h1>
        <h2> {count} </h2>
        <button onClick={handleAdd}> +1 </button>
        <button onClick={handleSubstract}> -1 </button>
        <button aria-label='btn-reset' onClick={handleReset}> Reset </button>
    </>
  )
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};