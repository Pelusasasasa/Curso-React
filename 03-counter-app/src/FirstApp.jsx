import PropTypes from 'prop-types';

export const FirstApp = ({title,subtitle}) => {


    return(
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
        </>    
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
};


FirstApp.defaultProps = {
    // title:"Hola, Soy Agustin",
    subtitle: 'Soy un subtitulo'
};