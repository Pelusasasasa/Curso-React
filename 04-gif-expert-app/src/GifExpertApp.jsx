import { useState } from "react"
import { AddCategory,GifGrid } from "./components";

export const GifExpertApp = () => {
    
    const [categories, setCategories] = useState(['Boca Juniors']);

    const onAddCategory = (category) => {

        if (categories.includes(category)) return;

        setCategories([category,...categories]);
    };

    const resetCategory = () =>{
        setCategories([]);
    }

  return (
    <>
        {/*Titulo*/}
        <h1>GifExpertApp</h1>

        {/*Input*/}
        <AddCategory 
            // setCategories={setCategories}
            onNewCategory={onAddCategory}
            resetCategory={resetCategory}
        />

        { categories.map( category => (
            <GifGrid 
                key={category}
                category={category} 
            />
        ))}
    </>


  )
}