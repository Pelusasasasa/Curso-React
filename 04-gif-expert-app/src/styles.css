* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(234, 234, 234);
  }
  
  body {
    padding: 60px;
  }

  .form{
    display: grid;
    grid-template-columns: 3fr 1fr;
    justify-content: space-around;
    gap: 20px;
  }

  button{
    border: none;
    background-color: greenyellow;
    border-radius: 50px;
    font-size: 20px;
    
  }

  button:hover{
    cursor: pointer;
    transform: scale(1.2);
  }
  
  input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 90%;
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 3rem;
    margin-bottom: 5px;
  }
  
  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
  
  .card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }
  
  .card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }