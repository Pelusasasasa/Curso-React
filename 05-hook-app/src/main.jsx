import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { HooksApp } from './HooksApp'
import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
// import { CounterApp } from './01-useState/CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterWithCustomHook/>
    {/* <HooksApp /> */}
    {/* <CounterApp value={0}/> */}
    
  </React.StrictMode>,
)
