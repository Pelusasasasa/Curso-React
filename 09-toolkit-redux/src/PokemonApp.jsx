import { useEffect } from "react"
import { useDispatch, useSelector } from "react-redux"
import { getPokemons } from "./store/slices/pokemon/thunks"

export const PokemonApp = () => {

  const dispatch = useDispatch();
  const {page, pokemons = [], isLoading} = useSelector( state => state.pokemons);

  useEffect(() => {

    dispatch( getPokemons() );
    
  }, []);
  
  return (
    <>
        <h3>Pokemon App</h3>
        <hr />
        { <span>Loading: { !isLoading ? 'True' : 'False'  }</span> }

        <ul>
            {pokemons.map((pokemon) => (
              <li key={pokemon.name}>{pokemon.name}</li>
            ))}
        </ul>
        <button onClick={() => dispatch( getPokemons(page) )}>Next</button>
    </>
  )
}
