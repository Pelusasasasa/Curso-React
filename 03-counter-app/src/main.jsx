import React from 'react';
import ReactDOM from 'react-dom/client';
import {App} from './HelloWordApp';
import { FirstApp } from './FirstApp';
import './style.css';
import {CounterApp} from './CounterApp';




ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={5} />
        {/* <FirstApp title='Hola Soy Vegeta'/> */}
    </React.StrictMode>
)
