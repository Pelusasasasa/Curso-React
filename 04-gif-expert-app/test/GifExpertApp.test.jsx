import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe(' Pruebas en el GifExpertApp', () => {

    test('Hacer match con el snapchot', () => {

        const {container} =  <GifExpertApp />;

        expect(container).toMatchSnapshot();
        
    });

});