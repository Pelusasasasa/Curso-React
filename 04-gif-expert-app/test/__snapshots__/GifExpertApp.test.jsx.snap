// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en el GifExpertApp Hacer match con el snapchot 1`] = `undefined`;
